### PR TITLE
Dependency list clean up (ui & presentation packages)

### PR DESCRIPTION
**Pull request by calebmshafer**
- Updated and ordered the dependency list of all presentation and ui packages
- Cleaned up the UI packages barrel file documentation sections

### CODE REVIEW

**Review by grigasp — APPROVED** [review]

File: ui/appui-react/src/test/TestUtils.ts [hunk 1]
```diff
@@ -30,7 +28,9 @@ const initialState: SampleAppState = {
   placeHolder: false,
 };
 
-/** @internal */
+/**
+ *
+ */
 export interface RootState {
```
Comment (roluk) [inline ui/appui-react/src/test/TestUtils.ts:34]: This seems like an unintentional change

**Review by roluk — CHANGES_REQUESTED** [review]

File: ui/appui-react/src/test/TestUtils.ts (hunk 1 above)
Comment (calebmshafer) [reply ui/appui-react/src/test/TestUtils.ts:34]: Ah good catch, auto-formatter added that back. I'll revert

Comment (calebmshafer) [reply ui/appui-react/src/test/TestUtils.ts]: fixed

Comment (roluk) [reply ui/appui-react/src/test/TestUtils.ts]: `RootState` interface is now neither `@public`, nor `@internal`, but I'm not sure this matters

**Review by roluk — APPROVED** [review]

File: ui/appui-react/src/test/TestUtils.ts (hunk 1 above)
Comment (calebmshafer) [reply ui/appui-react/src/test/TestUtils.ts]: It's a test utility so didn't think it mattered too much.

**Review by bsteinbk — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

**Review by ColinKerr — APPROVED** [review]

**Review by ramanujam-raman — APPROVED** [review]

**Review by bbastings — APPROVED** [review]

**Pull request merged**